### ISSUE TITLE
Aliases (Paths) Not Being Replaced

### ISSUE THREAD

**Issue by doitadrian** [swc-project/swc#2523]
**Describe the bug**
Just started playing with SWC and I I'd just like to say this is an amazing tool. Just amazed how it's fast and how much work has been put into it.  🙏🏻 

So, in my code, I have `~/........` import statements, but, once the SWC has done its magic, these are replaced with correct paths -  only when I set `"type": "commonjs"` in my `.swcrc` file. Otherwise, they just stay the same.

**Input code**

![image](https://user-images.githubusercontent.com/5121148/138599861-6cede9e7-5b0f-4247-b633-ca5f55bf684e.png)

So, this is the resulting code. As we can see, the import paths haven't changed (**~** is still here).

![image](https://user-images.githubusercontent.com/5121148/138599742-b23ac3a4-4bb6-4526-91cb-65d6e772a2b4.png)

But, if I were to use the commonjs module type, then I'd see the path built correctly:

![image](https://user-images.githubusercontent.com/5121148/138599885-a3861d3a-b1a2-442b-877b-551100f19379.png)

**Config**

`commonjs` works (paths are changed correctly):

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2020",
    "paths": {
      "~/*": ["./src/*"]
    },
    "baseUrl": "."
  },
  "module": {
    "type": "commonjs"
  }
}
```

This one does not work:

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2020",
    "paths": {
      "~/*": ["./src/*"]
    },
    "baseUrl": "."
  }
}
```

**Expected behavior**
The paths to be correctly replaced.

**Version**
The version of @swc/core: `1.2.102`

**Additional context**
Checked a couple of issues reported by other users (https://github.com/swc-project/swc/issues/2126, https://github.com/swc-project/swc/issues/2050), those didn't provide any clues unfortunately.

**Comment by ckken** [comment]
amd is no work too

**Comment by eturino** [comment]
es6 also doesn't work

**Comment by doitadrian** [comment]
Hey everybody, I saw a PR that was dealing with this was merged, so I went ahead and tried the latest `1.2.108` SWC version.

Unfortunately, I have to report that, at least on my end, the paths are still not being transformed the way they should be.

The steps I did were the following...

1. Created a brand new project and installed SWC.
2. Created the following files:

```
src
├── index.ts
└── utils
    └── index.ts
```

The `index.ts` looks like the follwoing:

```ts
import test from "~/utils";

test();
```

3. I created the same `.swcrc` config I used above:

```
{
  "jsc": {
    "parser": {
      "syntax": "typescript"
    },
    "target": "es2020",
    "paths": {
      "~/*": ["./src/*"]
    },
    "baseUrl": "."
  }
}
```

4. Ran `swc src -d dist --source-maps --copy-files --extensions '.ts'`

Ultimately, the results were the same as reported initially.

The compiled code was:

```
import test from "~/utils";
test();

//# sourceMappingURL=index.js.map
```

So, "~" wasn't replaced.

And, as stated above, once I switched to `commonjs` via ...

```  "module": {
    "type": "commonjs"
  }
```

... the transformations were successful.
